homeassistant/components/verisure/camera.py

```diff
@@ -21,29 +22,37 @@ def setup_platform(
     discovery_info: dict[str, Any] | None = None,
 ) -> None | Literal[False]:
     """Set up the Verisure Camera."""
-    if not int(hub.config.get(CONF_SMARTCAM, 1)):
+    coordinator = hass.data[DOMAIN]
+    if not int(coordinator.config.get(CONF_SMARTCAM, 1)):
         return False
 
     directory_path = hass.config.config_dir
     if not os.access(directory_path, os.R_OK):
```

Comment:
This is a weird check. Is it really needed? I'd assume we have read and write access to the config dir.

Reply:
I agree, this is weird logic, and there is way more weird logic; as this is all known by the API as well.

So for this part of the transition; I decided to leave in there until implementing the config flow (where this logic, and similar ones everywhere) will be removed.